go/vt/vtgate/evalengine/expr_logical.go

```diff
@@ -162,21 +207,18 @@ func (n *NotExpr) eval(env *ExpressionEnv) (eval, error) {
 
 func (n *NotExpr) typeof(env *ExpressionEnv) (sqltypes.Type, typeFlag) {
 	_, flags := n.Inner.typeof(env)
-	return sqltypes.Uint64, flags
+	return sqltypes.Int64, flags | flagIsBoolean
```

Comment:
These were just plain wrong before, it's actually an `Int64` empirically when testing against MySQL.